frontend/src/app/core/menu/user-menu/user-menu.component.html

```diff
@@ -22,7 +22,12 @@
     </ng-container>
     <ng-container *ngIf="showSignIn$ | async">
         <li class="nav-item">
-            <a [routerLink]="['/login']" routerLinkActive="active" class="nav-link">
+            <a
+                [routerLink]="['/login']"
+                routerLinkActive="active"
+                [queryParams]="(onLoginPage$ | async) ? null : { next: currentPath$ | async }"
```

Comment:
Prevents endless stacking of `?next=?next=?next=...` if a user (for some reason) keeps clicking on `Sign In`.